### PR TITLE
Update Cypress tests guidelines

### PR DESCRIPTION
**Pull request by nixocio**
Update Cypress tests guidelines

### CODE REVIEW

File: cypress/CYPRESS.md [hunk 1]
```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
...
+       cy.contains('.profile-list', 'John Doe');
+
+       // Asserting on elements that are expected to update
+       cy.get('.profile-status').should('contain', 'Profiles Loaded');
+     });
+   });
+   ```
+
+3. **Test Data**:
+   - Utilize dynamic data creation methods or fixtures to avoid dependency on hard-coded data.
```
Comment (akus062381) [inline cypress/CYPRESS.md:82]: fixtures should ideally only be used in component tests. E2E tests should communicate with the live API.

**Review by akus062381 — CHANGES_REQUESTED** [review]

**Review by akus062381 — APPROVED** [review]

File: cypress/CYPRESS.md [hunk 2]
```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
...
+
+   - Prefer using data attributes (like `data-cy`) for selecting elements.
+   - Avoid using CSS classes or IDs for element selection, as they can change frequently and are not unique.
+
+2. **Asynchronous Operations**:
+
+   - Properly handle the asynchronous nature of web applications, using Cypress's automatic waiting features or custom waits when necessary.
+   - Avoid to use `cy.wait` as much as possible - mainly with hardcoded wait values, prefer to use `cy.get` with assertions.
+
+   **Example**:
```
Comment (elyezer) [inline cypress/CYPRESS.md:55]: I really like that an example is being given, we could probably think about adding more examples related to the other items on the next iteration.

File: cypress/CYPRESS.md [hunk 3]
```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
...
+
+3. **Performance Considerations**:
+   - Keep tests efficient to minimize impact on the development and deployment process.
+
+### Documentation and Collaboration
+
+1. **Documentation of Custom Commands and Utilities**:
+
+   - Document any Cypress [custom commands](./support/core-commands.ts) or utilities developed for the project.
+   - Include usage examples and explanations of the purpose of each command.
```
Comment (elyezer) [inline cypress/CYPRESS.md:103]: Maybe link to a good example, or give an example here?

File: cypress/CYPRESS.md [hunk 4]
```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
...
+
+2. **Team Collaboration**:
```
Comment (elyezer) [inline cypress/CYPRESS.md:105]: Should we share a Slack channel for further conversations or related topics here?

File: cypress/CYPRESS.md (hunk 2 above)
Comment (nixocio) [reply cypress/CYPRESS.md:55]: @elyezer great suggestion, as we move forward we can add more details here.

File: cypress/CYPRESS.md (hunk 3 above)
Comment (nixocio) [reply cypress/CYPRESS.md:103]: @elyezer we are lacking this one. At least we do not have a dedicated markdown rendering this information.

**Pull request merged**